org.eclipse.jdt.debug.tests/tests/org/eclipse/jdt/debug/tests/core/VMInstallTests.java

```diff
@@ -201,6 +217,114 @@ public void testLibraryResolver4() throws Exception {
+	}
+
+	public void testJavaRuntimeQuerySystemPackages_nonModularJDK() throws Exception {
+		try (AutoCloseableSupplier<IVMInstall> nonModularVM = searchFirstNonModularVM()) {
+			IVMInstall vm = nonModularVM.get();
+
+			Set<String> packages = JavaRuntime.getProvidedVMPackages(vm, null);
+
+			LinkedHashSet<String> missing = new LinkedHashSet<>(COMMON_JAVA_PACKAGES);
+			missing.removeAll(packages);
+			assertTrue("Not all packages found in " + vm.getInstallLocation() + ", missing " + missing, missing.isEmpty());
+			assertTrue("Not found: " + NON_MODULAR_JDK_PACKAGE, packages.contains(NON_MODULAR_JDK_PACKAGE));
+			assertFalse("Found unexpected package: " + JAVA9_OR_LATER_PACKAGE, packages.contains(JAVA9_OR_LATER_PACKAGE));
+			assertFalse("Found unexpected package: " + JAVA11_OR_LATER_PACKAGE, packages.contains(JAVA11_OR_LATER_PACKAGE));
+
+			// Test that non-modular VMs ignore the release information
+			assertEquals(packages, JavaRuntime.getProvidedVMPackages(vm, "1.6"));
+			assertEquals(packages, JavaRuntime.getProvidedVMPackages(vm, "definitivly-not-a-version"));
+		}
+	}
+
+	private AutoCloseableSupplier<IVMInstall> searchFirstNonModularVM() throws IOException {
+		IVMInstallType vmType = JavaRuntime.getVMInstallType(StandardVMType.ID_STANDARD_VM_TYPE);
+
+		Set<File> candidateDirectories;
+		Path defaultVMLocation = JavaRuntime.getDefaultVMInstall().getInstallLocation().toPath();
+		int maxSearchDepth = 1;
+		Path root = Stream.iterate(defaultVMLocation, Objects::nonNull, Path::getParent) //
+				.skip(maxSearchDepth).findFirst().orElseThrow();
+		try (Stream<Path> directories = Files.walk(root, maxSearchDepth);) {
+			candidateDirectories = directories.map(Path::toFile).collect(Collectors.toCollection(HashSet::new));
+		}
+		Arrays.stream(JavaRuntime.getVMInstallTypes()).flatMap(t -> Arrays.stream(t.getVMInstalls())) //
+				.map(IVMInstall::getInstallLocation) //
+				.forEach(candidateDirectories::remove);
+
+		String nonModularJavaHome = System.getenv("NON_MODULAR_JAVA_HOME");
+		if (nonModularJavaHome != null) {
+			candidateDirectories.add(new File(nonModularJavaHome));
+			System.out.println(getName() + ": NON_MODULAR_JAVA_HOME: " + nonModularJavaHome);
+		}
+
+		System.out.println(getName() + ": found JVM candidates: " + candidateDirectories);
+
+		Optional<IVMInstall> candidateVm = Optional.empty();
+		for (File file : candidateDirectories) {
+			if (vmType.validateInstallLocation(file).isOK()) {
+				String id = "test-vm-" + System.nanoTime();
+				IVMInstall vmCandidate = vmType.createVMInstall(id);
+				vmCandidate.setInstallLocation(file);
+				if (!JavaRuntime.isModularJava(vmCandidate)) {
+					candidateVm = Optional.of(vmCandidate);
+					break;
+				}
+				vmType.disposeVMInstall(id);
+			}
+		}
+		assertTrue("No non-modular VM (<=Java-1.8) found.", candidateVm.isPresent());
+		IVMInstall vm = candidateVm.get();
+		System.out.println(getName() + ": found non-modular JVM: " + vm.getInstallLocation());
```

Comment:
I've added few system outs to print out what we've checked and what we finally found.